apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/controller/ConfigController.java

```diff
@@ -144,8 +155,42 @@ public ApolloConfig queryConfig(@PathVariable String appId, @PathVariable String
     }
 
     ApolloConfig apolloConfig = new ApolloConfig(appId, appClusterNameLoaded, originalNamespace,
-        mergedReleaseKey);
-    apolloConfig.setConfigurations(mergeReleaseConfigurations(releases));
+        latestMergedReleaseKey);
+
+    Map<String, String> latestConfigurations = mergeReleaseConfigurations(releases);
+
+    if (bizConfig.isConfigServiceChangeCacheEnabled()) {
```

Comment:
The logic for calculating configurationChanges should be placed in the IncrementalSyncConfigService. If the config cache is not enabled, data should be directly fetched from the database for comparison. If the config cache is enabled, data should be fetched from the cache for comparison, and the comparison results themselves can also be stored in the cache for reuse.